File: src/c1removeadjacentduplicates/Go/removeDuplicates.go
package main

// removeDuplicates function to remove consecutive duplicates from a string
func removeDuplicates(s string) string {
	stack := []rune{}

	for _, char := range s {
		if len(stack) > 0 && stack[len(stack)-1] == char {
			stack = stack[:len(stack)-1] // pop the last element
		} else {
			stack = append(stack, char) // push the current character
		}
	}

	return string(stack)
}

File: src/c1removeadjacentduplicates/JS/removeDuplicates.js
// removeDuplicates function to remove consecutive duplicates from a string
function removeDuplicates(s) {
    const stack = [];

    for (const char of s) {
        if (stack.length > 0 && stack[stack.length - 1] === char) {
            stack.pop(); // pop the last element
        } else {
            stack.push(char); // push the current character
        }
    }

    return stack.join('');
}

module.exports = removeDuplicates;
